FEAT: 회원가입/로그인을 구현한다.

deehyeon opened the pull request:
## 💡 관련 이슈
> 관련된 이슈 번호를 적어주세요


## 📝 작업 내용
Member와 Company 도메인, 엔티티를 구성했습니다.
헥사고날 아키텍처에 맞게 구성하려니 어렵네요.. 

헥사고날에서는 도메인(Aggregate Root)은 순수한 자바 객체로, 영속화 전략(JPA)은 어댑터(out/persistence) 쪽에 두어야 유지보수에 용이하다고 해서 이에 맞게 코드를 구성했습니다. 


## 🧑‍💻 변경 사항
추후에 테스트 코드와 관련 로직을 추가할 예정입니다.


## 💬 리뷰 요구사항(선택)
> 코드리뷰 부탁드립니다..

deehyeon committed:
#22 FEAT: Project 엔티티를 생성한다

deehyeon committed:
FEAT: 폴더 구조 변경

# Conflicts:
#	src/main/java/hanieum/conik/adapter/user/email/SMTPEmailSender.java
#	src/main/java/hanieum/conik/application/user/EmailCertService.java
#	src/main/java/hanieum/conik/application/user/EmailSender.java
#	src/main/java/hanieum/conik/application/user/required/EmailSender.java
#	src/main/java/hanieum/conik/user/application/required/EmailSender.java
#	src/test/java/hanieum/conik/domain/user/service/EmailServiceImplTest.java

deehyeon committed:
FEAT: 폴더 구조 변경

deehyeon committed:
FEAT: User 도메인과 엔티티 생성

deehyeon committed:
FEAT: User 도메인과 엔티티 생성

deehyeon committed:
FEAT: Company 도메인과 엔티티 생성

deehyeon committed:
FEAT: Company 도메인과 엔티티 생성

SeongHo5356 commented on src/main/java/hanieum/conik/adapter/company/persistence/CompanyEntity.java:
```diff
@@ -0,0 +1,53 @@
+@Table(name = "company")
+@Data
+@NoArgsConstructor(access = AccessLevel.PROTECTED)
+public class CompanyEntity extends BaseEntity {
```
도메인 수준에서 사용하는 모델과 peresistence 레벨에서 사용하는 엔티티를 구분하지 않고 하나로 사용할거라면 Company로만 이름을 짓는게 좋을 것 같아요. BaseEntity 활용하는 거 너무 좋네요

```suggestion
public class Company extends BaseEntity
```

SeongHo5356 commented on src/main/java/hanieum/conik/adapter/company/persistence/CompanyEntity.java:
```diff
@@ -0,0 +1,53 @@
+    @Column(name = "company_profile_url", nullable = false)
+    private String profileUrl;
+
+    @Enumerated(EnumType.STRING)
+    @Column(name = "company_status", nullable = false)
+    private CompanyStatus status;
```
저는 개인적으로 나중에 변수를 불러쓰거나 할 때 같은 엔티티속성명이 존재하면 헷갈리는 일이 많아서 prefix로 엔티티명을 사용하는 데 어떻게 생각하시나요?
참고로 자동으로 컬럼명은 스네이크 케이스로 만들어줘서, 같은 걸로 하고 싶다면 엔티티속성명은 지정하지 않아도 돼요
```suggestion
@Enumerated(EnumType.STRING) 
@Column(nullable = false)
private CompanyStatus companyStatus
```

deehyeon replied:
변수를 불러쓸때 company.getCompanyStatus() 이렇게 하면 변수명이 길어지거나 할때 코드가 너무 길어지는 것 같다고 생각해서 저는 company.getStatus() 만으로도 의미가 잘 전달된다고 생각해서 계속 저렇게 작성하기는 했습니다.
헷갈리는 경우 방지하려면 prefix로 해도 좋은 것 같아요. 그렇게 통일할까요?

SeongHo5356 replied:
prefix 빼고하는걸로 결정!

SeongHo5356 commented on src/main/java/hanieum/conik/adapter/user/email/dto/AuthCodeRequest.java:
아 이렇게 도메인별 패키지 안에서 나누는게 아니고, 패키지 안에서 도메인마다 나누는 것이군요

SeongHo5356 commented on src/main/java/hanieum/conik/adapter/user/persistence/UserEntity.java:
```diff
@@ -0,0 +1,40 @@
+@Data
+@Table(name = "User")
+@NoArgsConstructor(access = AccessLevel.PROTECTED)
+public class UserEntity extends BaseEntity {
```
위에 의견과 동일해요
아마 근데 User로 만들면 JWT에서 사용하는 내장 User와 겹쳐서 문제가 생길거라 member나 다른 걸로 변경해야겠군요
```suggestion
public class User extends BaseEntity
```

deehyeon replied:
넵 일단 초기에 User로 되어있어서 그대로 가져가기는 했는데 member로 수정하는게 좋을 것 같아요

SeongHo5356 commented on src/main/java/hanieum/conik/adapter/user/persistence/UserEntity.java:
```diff
@@ -0,0 +1,40 @@
+@NoArgsConstructor(access = AccessLevel.PROTECTED)
+public class UserEntity extends BaseEntity {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
```
위에는 한 줄에 정리했는데, 여기서는 두 줄에 해서 이런 것도 일관성 있게 가져가면 좋을 것 같아요

SeongHo5356 replied:
이후에 XML로 분리하는 거로 결정!

SeongHo5356 commented on src/main/java/hanieum/conik/adapter/user/persistence/UserEntity.java:
```diff
@@ -0,0 +1,40 @@
+    @Column(name = "email", nullable = false, unique = true, length = 100)
+    private String email;
+
+    @Column(name = "phone_number", nullable = false, length = 20)
```
위에서 말했듯이 기본적으로 스네이크 케이스로 컬럼명은 생성해줘요!

SeongHo5356 commented on src/main/java/hanieum/conik/application/user/EmailSender.java:
```diff
@@ -1,4 +1,4 @@
-package hanieum.conik.user.application.required;
+package hanieum.conik.application.user;
```
음 혹시 required 패키지를 뺀 이유가 있을까요 ?

deehyeon replied:
폴더 구조 수정하다가 빠진 것 같아요! 다시 만들어두겠습니다.

SeongHo5356 commented on src/main/java/hanieum/conik/domain/company/Company.java:
```diff
@@ -0,0 +1,27 @@
+package hanieum.conik.domain.company;
+
+public class Company {
```
오 이걸 못봤네요 도메인 객체를 따로 나눴는데 그 이유가 궁금해요 저도 이렇게 할까 아니면, 그냥 엔티티를 활용할까 고민하다가 
1. 토비의 강의에서 꼭 나눠야 하는 것은 아니라고도 했고, 
2. 이렇게 하면 objectMapper를 하나하나 구현해야 해서 개발 시간이 너무 길어질 것 같아서

그냥 엔티티만 활용하는 방향으로 구현하고 있거든요!

deehyeon replied:
도메인에서는 순수 자바 코드로만 작성하는 것이 좋다고해서 나누어 구성하긴 했습니다.
근데 진짜 순수 도메인만 남겨두려니까 dto도 도메인 폴더가 아닌 adapter/webapi/dto 이런 식으로 adapter 폴더에 구성해야하더군요.. 
고민하다가 우선 분리해두긴 했는데, 제 생각에도 엔티티를 활용하는게 좋을 것 같아요

SeongHo5356 replied:
엔티티만 활용하는 걸로 결정!

SeongHo5356 commented on src/main/java/hanieum/conik/domain/user/dto/UserSignUpRequest.java:
```diff
@@ -0,0 +1,10 @@
+                                String password,
+                                String phoneNumber,
+                                Boolean termsOfServiceAgreed,
+                                UserRole role) {}
```
```suggestion
public record UserSignUpRequest(
    @Email 
    String email,
                               
    String password,

    String phoneNumber,

    Boolean termsOfServiceAgreed,

    UserRole role
) {}
```
이런식으로 작성하면 가독성이 좀 더 좋을 것 같아요

SeongHo5356 reviewed (commented):
굳굳 좋습니다~! 
변수명이나, 필드명 짓는 거 관련해서 모두 통일하면 좋을 것 같아서 그 부분 다음에 논의 해보면 좋을 것 같네요.
추가로, 토비의 강의에서는 in/out 보다는 required/provided를 사용하더라구요 이번 프로젝트 과정에서는 강의를 기준으로 정하고 구현해보고 추후에 헥사고날의 개념을 좀 더 생각하면서 리팩토링해보면 좋을 것 같아요

SeongHo5356 committed:
#25 Feat: 공장 견적서 엔티티를 작성한다

SeongHo5356 committed:
#25 Feat: 공고 엔티티를 작성한다

SeongHo5356 committed:
#25 Refactor: enum 값들에 설명을 추가한다

SeongHo5356 committed:
#25 Refactor: enum 값들에 설명을 변경한다

deehyeon committed:
#27 Feat: Member와 Company 도메인 코드와 엔티티 코드 병합

deehyeon committed:
#27 Feat: Member와 Company 도메인 코드와 엔티티 코드 병합

SeongHo5356 commented on src/main/java/hanieum/conik/domain/company/Company.java:
```diff
@@ -1,5 +1,7 @@
 
+import hanieum.conik.domain.company.enumerate.CompanyStatus;
+
 public class Company {
```
도메인모델을 삭제하고 엔티티만 활용하기로 결정한 걸로 저는 생각했었는데, 이건 도메인 모델 같아요

SeongHo5356 commented on src/main/java/hanieum/conik/domain/member/Member.java:
```diff
@@ -0,0 +1,69 @@
+            throw new UserException(UserErrorType.INVALID_PASSWORD);
+        }
+        this.password = newPassword;
+    }
```
여기에 위치시키는 거 좋습니다~!!

SeongHo5356 reviewed (commented):
도메인 객체 없애고 엔티티만 남겨서 다시 올려주세요~!

SeongHo5356 committed:
#25 Refactor: jpa annotaion을 xml로 분리하여 도메인 로직의 가독성을 높인다.

SeongHo5356 committed:
#25 Feat: 공통으로 사용하는 id를 AbstracEntity로 관리한다

SeongHo5356 committed:
#25 Refactor: 패키지 구조를 재정의한다

SeongHo5356 committed:
#25 Feat: orm들을 등록하고 관리한다

SeongHo5356 committed:
#25 Feat: 기업에게 요청하는 정보를 담은 ProjectRequest엔티티를 작성한다

SeongHo5356 committed:
#25 Refactor: 테스토용 환경변수를 분리해서 관리한다

SeongHo5356 committed:
#25 Refactor: 테스트 환경에서 H2 DB를 사용한다

SeongHo5356 committed:
#25 Refactor: h2 DB 관련 의존성을 추가한다

SeongHo5356 committed:
#25 Fix: CI 환경에서 APPLICATION_YML을 사용한다

SeongHo5356 committed:
#25 Feat: 거래 상태 관련 엔티티를 작성한다

SeongHo5356 committed:
#25 Refactor: 헥사고날 아키텍쳐에 맞게 패키지 구조를 정리한다

SeongHo5356 committed:
#25 Test: ProposalItem 고아객체 삭제 테스트를 작성한다

SeongHo5356 committed:
#25 Refactor: 워크 플로우 적용 브랜치를 변경한다

SeongHo5356 committed:
#25 Feat: 거래상태 도메인 엔티티의 jpa 에노테이션들을 xml로 관리한다

deehyeon committed:
#27 Feat: Company 도메인 코드와 엔티티 코드 병합

deehyeon committed:
#27 Feat: Company 도메인 코드와 엔티티 코드 병합

deehyeon committed:
#27 Feat: Company Status 생성

deehyeon committed:
 Conflicts:
	src/main/java/hanieum/conik/application/member/required/EmailSender.java
	src/main/java/hanieum/conik/application/user/EmailSender.java
	src/main/java/hanieum/conik/application/user/required/EmailSender.java

deehyeon committed:
#27 Feat: User 도메인 이름을 Member로 변경

deehyeon committed:
#27 Feat: Email 형식 증명 코드 추가

deehyeon committed:
#27 Feat: Email 형식 증명 코드 추가

deehyeon committed:
#27 Feat: 회원가입시 이메일 검증 예외 처리

deehyeon committed:
#27 Feat: 회원가입시 이메일 검증 예외 처리

deehyeon committed:
#27 Feat: 회원가입/로그인 관련 dto, 에러 타입 추가

deehyeon committed:
#27 Feat: 회원가입/로그인 관련 dto, 에러 타입 추가

# Conflicts:
#	src/test/java/hanieum/conik/domain/member/service/EmailServiceImplTest.java

deehyeon committed:
#27 Feat: MemberRepository와 MemberService 생성

deehyeon committed:
#27 Feat: MemberRepository와 MemberService 생성

deehyeon committed:
#27 Feat: Security, Jwt 관련 의존성 추가

deehyeon committed:
#27 Feat: Security, Jwt 관련 의존성 추가

deehyeon committed:
#27 Feat: SecurityConfig와 CorsConfig 설정

deehyeon committed:
#27 Feat: SecurityConfig와 CorsConfig 설정

deehyeon committed:
#27 Feat: CustomUserDetail과 CurrentUser 인터페이스 생성

deehyeon committed:
#27 Feat: CustomUserDetail과 CurrentUser 인터페이스 생성

deehyeon committed:
#27 Feat: JwtTokenProviderPort 인터페이스와 JwtTokenProvider 구현체 생성, JwtAuthenticationFilter 구현

deehyeon committed:
#27 Feat: JwtTokenProviderPort 인터페이스와 JwtTokenProvider 구현체 생성, JwtAuthenticationFilter 구현

deehyeon committed:
#27 Feat: refreshtoken 관련 인터페이스와 구현체, 서비스 코드 구현

deehyeon committed:
#27 Feat: refreshtoken 관련 인터페이스와 구현체, 서비스 코드 구현

deehyeon committed:
#27 Feat: 회원가입,로그인 관련 로직 구현

deehyeon committed:
#27 Feat: 회원가입,로그인 관련 로직 구현

deehyeon committed:
#27 Feat: ConikApplication에 @EnableJpaAuditing 어노테이션 추가

deehyeon committed:
#27 Feat: ConikApplication에 @EnableJpaAuditing 어노테이션 추가

SeongHo5356 commented on src/main/java/hanieum/conik/adapter/member/webapi/AuthController.java:
```diff
@@ -0,0 +1,46 @@
+    - 이메일, 비밀번호 등의 정보를 입력하여 회원가입합니다.
+    - 회원가입 완료 시 로그인도 완료됩니다.
+    """)
+    @PostMapping("/signup")
+    public ResponseEntity<ApiResponse<MemberLoginResponse>> signUp(@RequestBody MemberSignUpRequest request) {
```
여기도 @Valid 달아야겠다!

deehyeon replied:
엇 저거 빼먹었구나 감사합니다

SeongHo5356 commented on src/main/java/hanieum/conik/application/member/provided/Auth.java:
```diff
@@ -0,0 +1,15 @@
+import hanieum.conik.domain.member.Member;
+import jakarta.validation.Valid;
+
+/**
+ * 회원가입/로그인 로직을 구현한다.
+ */
```
강의에서도 이런식으로 주석처리했던 것 같은데 디테일이 살이있네요

SeongHo5356 commented on src/main/java/hanieum/conik/application/member/provided/findMember.java:
```diff
@@ -0,0 +1,10 @@
+/**
+ * 회원을 조회한다
+ */
+public interface findMember {
```
MemberFinder로 이름을 짓는건 어떨까요?

SeongHo5356 commented on src/main/java/hanieum/conik/application/member/required/MemberRepository.java:
```diff
@@ -0,0 +1,14 @@
+
+    Optional<Member> findByEmail(Email email);
+
+    Optional<Member> findById(Long memberId);
```
해당 메서드는 선언할 필요없이 자동으로 제공해줄거에요 아래 관련 내용 첨부해놓을게요

[JPA 기본 구현 메서드](https://frogand.tistory.com/22)

SeongHo5356 commented on src/main/java/hanieum/conik/application/member/AuthService.java:
```diff
@@ -0,0 +1,67 @@
+        Member member = memberRepository.findByEmail(new Email(request.email()))
+                .orElseThrow(() -> new AuthException(AuthErrorType.MEMBER_NOT_FOUND));
+
+        if (member.getPassword().equals(request.password())) {
```
비밀번호는 DB에 저장 시 단방향 해시로 암호화해서 저장하고, 해시값을 비교하는 방식으로 검증하면 좋을 것 같아요

deehyeon committed:
#27 Feat: MemberSignUpRequest에 @Valid 인증 추가

deehyeon committed:
#27 Feat: MemberSignUpRequest에 @Valid 인증 추가

SeongHo5356 commented on src/main/java/hanieum/conik/application/member/AuthService.java:
```diff
@@ -0,0 +1,67 @@
+
+        checkDuplicateEmail(request);
+
+        Member member = Member.signUp(new Email(request.email()), request.password(), request.phoneNumber(), request.termsOfServiceAgreed(), request.role());
```
저번 pr에서 나왔던 주제 같은데, 결국 dto를 사용하지 않았네요. 이유를 알 수 있을까요?

deehyeon replied:
dto 사용하는걸로~

SeongHo5356 commented on src/main/java/hanieum/conik/application/member/AuthService.java:
```diff
@@ -0,0 +1,67 @@
+
+        memberRepository.save(member);
+
+        return login(new MemberLoginRequest(member.getEmail().address(), member.getPassword()));
```
이메일을 따로 VO로 다뤄야할 생각하시는 이유가 뭔가요?

deehyeon replied:
dto에서 @Email을 사용해서 검증을 해주긴 하지만,
나중에 어딘가에서 DTO를 거치지 않고 도메인 객체를 생성하게 되는 로직이 생긴다면 이때 유효성 검증이 들어가야 타입 안정성을 유지할 수 있다고 생각해서 VO를 따로 만들어 구성했습니다!

SeongHo5356 commented on src/main/java/hanieum/conik/domain/company/enumerate/CompanyStatus.java:
```diff
@@ -0,0 +1,7 @@
+public enum CompanyStatus {
+    PENDING,
+    APPROVED,
+    REJECTED
```
지금 상태가 어떤 용도의 상태인지 알 수 있게 아래 처럼 바꾸는 건 어떤가요?

```suggestion
    REGISTER_PENDING,
    REGISTER_APPROVED,
    REGISTER_REJECTED
```

SeongHo5356 commented on src/main/java/hanieum/conik/domain/company/Company.java:
```diff
@@ -0,0 +1,52 @@
+    @Column(nullable = false)
+    private String owner;
+
+    @Column(nullable = false)
+    private String ownerEmail;
+
+    @Column(nullable = false)
+    private String ownerPhoneNumber;
```
사람의 이메일과 휴대폰 번호는 Member에서 가지고 있고 Company는 멤버가 속한 회사를 나타내는 거다보니 회사이메일과 회사 전화번호로 하는 게 어떨까요?
만약 그렇게 변경한다면, 컬럼명도 변경되어야할 것 같아요

deehyeon replied:
요부분은 유저 플로우 로직 수정이 필요해보여 우선 디스코드에 올려두었습니다!

SeongHo5356 commented on src/main/java/hanieum/conik/domain/company/Company.java:
```diff
@@ -0,0 +1,52 @@
+    private String registrationCertificateUrl;
+
+    @Column(nullable = false)
+    private String bankAccountUrl;
```
통장사본일테니 BankbookCopy 어떤가요?

SeongHo5356 commented on src/main/java/hanieum/conik/domain/member/enumerate/MemberRole.java:
```diff
@@ -0,0 +1,7 @@
+    OWNER,
+    STAFF,
+    INDIVIDUAL
+}
```
```suggestion
@Getter
@RequiredArgsConstructor
public enum MemberRole {
    OWNER("기업 대표"),
    STAFF("기업 근로자"),
    INDIVIDUAL("소상공인")

    private final String description;
}
```

SeongHo5356 commented on src/main/java/hanieum/conik/domain/member/exception/VerifyEmailException.java:
```diff
@@ -0,0 +1,21 @@
+
+import hanieum.conik.global.apiPayload.exception.GlobalException;
+
+public class VerifyEmailException extends GlobalException {
```
음 ErrorType은 UserErrorType에 있는 거를 사용하면서 VerifyEmailException은 따로 분리해서 만들면 코드의 일관성이 좀 떨어질 수도 있겠다고 생각이 들어요. Email을 VO로 다루고 있긴 하지만, UserException을 통해서 user도메인의 에러는 모두 처리하는 게 어떨까요?
추가로, Member로 패키지명을 바꾼 만큼, userException에서 memberException으로 이름도 바꿔주면 좋을 것 같아요

SeongHo5356 commented on src/main/java/hanieum/conik/global/adapter/redis/RefreshTokenAdapter.java:
```diff
@@ -0,0 +1,29 @@
+
+@Component
+@RequiredArgsConstructor
+public class RefreshTokenAdapter implements RefreshTokenPort {
+
+    private final RefreshTokenService refreshTokenService;
+
+
+    @Override
+    public void saveRefreshToken(Long memberId, String refreshToken, long expirationMinutes) {
+        refreshTokenService.saveRefreshToken(memberId, refreshToken, expirationMinutes);
+    }
+
+    @Override
+    public String findRefreshToken(Long memberId) {
+        return refreshTokenService.getRefreshToken(memberId);
+    }
+
+    @Override
+    public void deleteRefreshToken(Long memberId) {
+        refreshTokenService.deleteRefreshToken(memberId);
+    }
+}
```
음 지금 Adapter에서 Service를 의존하고 있는게 방향이 잘못된 것 같아요. Adapter는 Port의 구현체이기 때문에, 단순히 redis에 저장, 추출 등의 역할만 수행하고 그런 기능을 RedisService에서 가져와서 사용하는 식으로 바뀌어야할 것 같아요. 
제가 생각하는 최고의 방향은 RefreshTokenAdpater과 RedisTokenService를 합치고, 요 합친 어댑터를 회원가입이나, 로그인 비즈니스 로직에 가져다가 쓰는게 좋아보이네요

deehyeon replied:
RefreshTokenAdpater 을 없애고 RedisMemoryMap을 사용하는 걸로!!!

SeongHo5356 commented on src/main/java/hanieum/conik/global/adapter/security/jwt/JwtTokenProvider.java:
```diff
@@ -0,0 +1,119 @@
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class JwtTokenProvider implements JwtTokenProviderPort {
```
redis도 이런 느낌이 되면 좋을 것 같아요

SeongHo5356 commented on src/main/java/hanieum/conik/global/config/CorsConfig.java:
```diff
@@ -0,0 +1,30 @@
+
+import java.util.List;
+
+@Configuration
+public class CorsConfig {
+
+    @Bean
+    public CorsConfigurationSource corsConfigurationSource() {
+        CorsConfiguration config = new CorsConfiguration();
+        config.setAllowedOrigins(List.of(
+                "http://localhost:8080"
+        ));
+        config.setAllowedMethods(List.of("GET","POST","PUT","DELETE","OPTIONS"));
+        config.setAllowedHeaders(List.of("*"));
+        config.setAllowCredentials(true);
+        config.setMaxAge(3600L);
+
+        UrlBasedCorsConfigurationSource source = new UrlBasedCorsConfigurationSource();
+        source.registerCorsConfiguration("/v1/**", config);
+        return source;
+    }
+}
+
```
이 내용은 이미 main branch에 WebMvcConfig로 구현돼 있을 거에요!

SeongHo5356 reviewed (changes requested):
고생하셨습니다!! 구현된 에러타입들을 보니 정말 꼼꼼하게 구현하신 것 같아요. 가장 고민해봐야할 건 Redis를 구현한 부분을 port,Adapter로 분리한 부분일 것 같아요! 리뷰가 많은데 좀만 더 화이팅입니다~!

SeongHo5356 committed:
#25 Refactor: orm 중복 내용을 삭제한다

SeongHo5356 committed:
Merge pull request #29 from HANIEUMICT/feat/#25-quote

FEAT: 공고, 입찰, 거래상태 도메인의 엔티티를 생성한다

deehyeon committed:
#27 Fix: dto 사용으로 변경

deehyeon committed:
#27 Fix: class에 맞게 클래스 이름 변경

deehyeon committed:
#27 Fix: MemberRole 각 값에 설명 추가

deehyeon committed:
#27 Fix: 중복되는 redis 관련 코드를 MemoryMap 인터페이스 구현체로 통합

deehyeon committed:
#27 Fix: User 를 Member로 변경

deehyeon committed:
#27 Fix: 중복되는 Config 파일 삭제

deehyeon committed:
#27 Fix: Jpa에서 기본으로 구현되는 메서드 삭제

deehyeon committed:
#27 Feat: CompanyStatus에 description 추가

deehyeon committed:
Merge remote-tracking branch 'origin/feat/login' into feat/login

deehyeon committed:
#27 Fix : 분리되어있던 Error 타입 병합

deehyeon committed:
#27 Fix : 통장 사본 필드명 변경

deehyeon committed:
#27 Fix : 비밀번호 해시화 저장으로 변경

SeongHo5356 commented on src/main/java/hanieum/conik/ConikApplication.java:
```diff
@@ -3,9 +3,11 @@
 
 @EnableFeignClients
 @SpringBootApplication
+@EnableJpaAuditing
```
이것때문에 현재 테스트 시에 에러가 나는 것 같은데, 이건 유지하고
ProposalOrphanRemovalTest 파일에서 
클래스에 붙은 @EnableJpaAuditing 이거 없애고 아래처럼 작성하면 잘 통과할거에요

```suggestion
@SpringBootTest
@Transactional
class ProposalOrphanRemovalTest {

    @Autowired
    private EntityManager em;

    @TestConfiguration
    @EnableJpaAuditing
    static class JpaAuditingTestConfig {}
```

SeongHo5356 commented on src/main/java/hanieum/conik/adapter/member/webapi/AuthController.java:
```diff
@@ -0,0 +1,46 @@
+    @PostMapping("/signup")
+    public ResponseEntity<ApiResponse<MemberLoginResponse>> signUp(@RequestBody @Valid MemberSignUpRequest request) {
+        MemberLoginResponse loginResponse = authService.register(request);
+        return new ResponseEntity<>(ApiResponse.success(loginResponse), HttpStatus.OK);
```
```suggestion
return ApiResponse.success(loginResponse)
```

SeongHo5356 commented on src/main/java/hanieum/conik/adapter/member/webapi/AuthController.java:
```diff
@@ -0,0 +1,46 @@
+    @PostMapping("/login")
+    public ResponseEntity<ApiResponse<MemberLoginResponse>> login(@RequestBody @Valid MemberLoginRequest request) {
+        MemberLoginResponse loginResponse = authService.login(request);
+        return new ResponseEntity<>(ApiResponse.success(loginResponse),HttpStatus.OK);
```
```suggestion
return ApiResponse.success(loginResponse)
```

SeongHo5356 commented on src/main/java/hanieum/conik/application/member/AuthService.java:
```diff
@@ -0,0 +1,75 @@
+        }
+    }
+
+    private void checkDuplicateEmail(MemberSignUpRequest signUpRequest){
+        if (memberRepository.findByEmail(new Email(signUpRequest.email())).isPresent()) {
+            throw new MemberException(MemberErrorType.EMAIL_DUPLICATE);
+        }
+    }
```
```suggestion
private void checkDuplicateEmail(MemberSignUpRequest signUpRequest) {
    boolean isDuplicate = memberRepository.findByEmail(new Email(signUpRequest.email())).isPresent();
    if (isDuplicate) {
        throw new MemberException(MemberErrorType.EMAIL_DUPLICATE);
    }
}
```

이렇게 하면 읽을 때 더 잘 이해할 수 있을 것 같긴 한데 너무 사소한 부분이라 이렇게 해야할까 고민도 되네요

deehyeon replied:
오호 이게 좀 더 가독성이 좋을 수 있겠네요
저는 메소드 이름에 checkDuplicateEmail이 있어서 if 부분이 중복 체크하는거라고 자연스럽게 생각할 수 있다고 생각해서 이렇게 올렸습니다.!!

SeongHo5356 reviewed (approved):
고생하셨습니다~👏
